### PR TITLE
Use '>=' for versions

### PR DESCRIPTION
**Pull request by MaximilianR**
So package upgrades don't cause breaks

### CODE REVIEW

**Comment by MaximilianR** [comment]
Resolves https://github.com/msiemens/PyGitUp/issues/44

**Comment by msiemens** [comment]
This will lead to complicated. Except for docopt, which uses semantic versioning, no package makes any claims about compatibility between releases (as far as I can see at least).

For docopt we definitely can adopt this version comparison, for the other ones I don't know what the best option is -- having the recent version at risk of running into incompatibilities, or always having to release a new version when the dependencies are updated but on the other hand not risking breakage.

**Comment by MaximilianR** [comment]
Yes those points make sense. Although, I think the biggest issue here is that the current model forbids people from upgrading their packages, which is not OK - particularly for something like `six`, which lots of other libraries use.

Other libraries that I work with (pandas, xarray), all do the `>=` model, and then fix breakages if they occur.

Separately, I'm not sure why the code _breaks_ at runtime, but that's more of a curiosity than anything else

**Comment by msiemens** [comment]
> But I think the biggest issue here is that the current model forbids people from upgrading their packages, which is not OK - particularly for something like six, which lots of other libraries depend on.

Point taken. The deeper issue is a lack of isolation of dependencies with Python applications. When working on Python projects, you usually use virtualenv to install dependencies locally without affecting other projects. Sadly, there is no such habbit with distributed Python applications. (Actually, there is an attempt called [pipsi](https://github.com/mitsuhiko/pipsi), but it's definitely not common practice). But in the end that's a cultural problem and propably we won't be able to change that.

Long story short: I'm merging this.

**Comment by msiemens** [comment]
Just for curiosity: This issue really highlights what Rust is doing in that regard: exported symbols (mainly functions AFAIK) always include a hash that is calculated based on the version number. So applications not only depend on a specific library, but a specific version as well, eliminating conflicts like encountered here.
